web/src/pill_typeahead.ts

```diff
@@ -36,25 +37,40 @@ export function set_up(
         user_source?: () => User[];
         exclude_bots?: boolean;
         update_func?: () => void;
+        help_on_empty_strings?: boolean;
+        display_pill?: (sub: StreamSubscription) => string;
     },
 ): void {
     if (!opts.user && !opts.user_group && !opts.stream) {
         blueslip.error("Unspecified possible item types");
         return;
     }
-    const include_streams = (query: string): boolean =>
-        opts.stream !== undefined && query.trim().startsWith("#");
+
+    opts.help_on_empty_strings ||= false;
+
     const include_user_groups = opts.user_group;
     const include_users = opts.user;
     const exclude_bots = opts.exclude_bots;
 
+    const include_streams: (query: string) => boolean | undefined = (query) => {
+        if (include_users || include_user_groups) {
+            return opts.stream && query.trim().startsWith("#");
+        }
```

Comment:
This might be better written by just having an explicit `stream_match_prefix` parameter that would be `#` for the callers that want such a prefix, which happens to be the ones that are only including streams? Seems like a more orthogonal/correct API.

Reply:
> This might be better written by just having an explicit stream_match_prefix parameter that would be # for the callers that want such a prefix, which happens to be the ones that are only including streams?

Callers which only include streams should have the `stream_match_prefix` value a s "" (empty string). And for the callers which include streams and users/user groups should have the `stream_match_prefix` value as "#".

Reply:
I am not sure I understand, but yeah I think that would be fine.